### PR TITLE
Add missing tidiers

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR adds missing tidy method for `step_intercept()` and `step_lag()`.

To close https://github.com/tidymodels/recipes/issues/730

### CODE REVIEW

File: R/lag.R [hunk 1]
```diff
@@ -20,6 +20,11 @@
 #' @details The step assumes that the data are already _in the proper sequential
 #'  order_ for lagging.
 #'
+#' # Tidying
```
Comment (simonpcouch) [inline R/lag.R:23]: I hadn't seen this `#' # L1 Header` in roxygen before! Nice!

File: R/lag.R [hunk 2]
```diff
@@ -20,6 +20,11 @@
...
+#'
+#' When you [`tidy()`][tidy.recipe()] this step, a tibble with column
+#' `terms` (the columns that will be affected) is returned.
```
Comment (simonpcouch) [inline R/lag.R:26]: I realize this is a boilerplate description from other tidiers, so don't want this to block the PR, but I'm having trouble demonstrating that `terms` column for myself. Where does that come up?

``` r
library(recipes)
#> Loading required package: dplyr
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
#> 
#> Attaching package: 'recipes'
#> The following object is masked from 'package:stats':
#> 
#>     step
  
n <- 10
start <- as.Date("1999/01/01")
end <- as.Date("1999/01/10")

df <- data.frame(
  x = runif(n),
  index = 1:n,
  day = seq(start, end, by = "day")
)

rec <-
  recipe(~., data = df) %>%
  step_lag(index, day, lag = 2:3) %>%
  step_intercept(value = 2)

rec %>% tidy()
#> # A tibble: 2 × 6
#>   number operation type      trained skip  id             
#>    <int> <chr>     <chr>     <lgl>   <lgl> <chr>          
#> 1      1 step      lag       FALSE   FALSE lag_4KHD3      
#> 2      2 step      intercept FALSE   FALSE intercept_quv0C

rec %>% prep() %>% tidy()
#> # A tibble: 2 × 6
#>   number operation type      trained skip  id             
#>    <int> <chr>     <chr>     <lgl>   <lgl> <chr>          
#> 1      1 step      lag       TRUE    FALSE lag_4KHD3      
#> 2      2 step      intercept TRUE    FALSE intercept_quv0C
```

<sup>Created on 2022-12-22 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Review by simonpcouch — APPROVED** [review]
Looks good! One question to aid my own understanding.

File: R/lag.R (hunk 2 above)
Comment (EmilHvitfeldt) [reply R/lag.R:26]: You need to do `tidy(1)` and `tidy(2)` to tidy specific steps.

``` r
library(recipes)

n <- 10
start <- as.Date("1999/01/01")
end <- as.Date("1999/01/10")

df <- data.frame(
  x = runif(n),
  index = 1:n,
  day = seq(start, end, by = "day")
)

rec <-
  recipe(~., data = df) %>%
  step_lag(index, day, lag = 2:3) %>%
  step_intercept(value = 2)

rec %>% tidy()
#> # A tibble: 2 × 6
#>   number operation type      trained skip  id             
#>    <int> <chr>     <chr>     <lgl>   <lgl> <chr>          
#> 1      1 step      lag       FALSE   FALSE lag_p9rbM      
#> 2      2 step      intercept FALSE   FALSE intercept_DXVUa
rec %>% tidy(1)
#> # A tibble: 2 × 2
#>   terms id       
#>   <chr> <chr>    
#> 1 index lag_p9rbM
#> 2 day   lag_p9rbM
rec %>% tidy(2)
#> # A tibble: 0 × 2
#> # … with 2 variables: terms <chr>, id <chr>

rec %>% prep() %>% tidy()
#> # A tibble: 2 × 6
#>   number operation type      trained skip  id             
#>    <int> <chr>     <chr>     <lgl>   <lgl> <chr>          
#> 1      1 step      lag       TRUE    FALSE lag_p9rbM      
#> 2      2 step      intercept TRUE    FALSE intercept_DXVUa
rec %>% prep() %>% tidy(1)
#> # A tibble: 2 × 2
#>   terms id       
#>   <chr> <chr>    
#> 1 index lag_p9rbM
#> 2 day   lag_p9rbM
rec %>% prep() %>% tidy(2)
#> # A tibble: 0 × 1
#> # … with 1 variable: id <chr>
```

<sup>Created on 2022-12-22 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Comment (EmilHvitfeldt) [reply R/lag.R:26]: Seeing this I'm seeing results that doesn't match. Let me fix that

Comment (simonpcouch) [reply R/lag.R:26]: Ah, I see! Okay, feel free to resolve when you feel this is good to go.

Comment (EmilHvitfeldt) [reply R/lag.R:26]: fixed now

**Pull request merged**